test/mocks/services/mock_upvote_repository_service.dart

```diff
@@ -0,0 +1,42 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:mockito/mockito.dart";
+import "package:proxima/models/database/firestore/id_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+import "package:proxima/models/database/vote/upvote_state.dart";
+import "package:proxima/services/database/upvote_repository_service.dart";
+
+class MockUpvoteRepositoryService<ParentIdFirestore extends IdFirestore>
+    extends Mock implements UpvoteRepositoryService<ParentIdFirestore> {
+  @override
+  Future<UpvoteState> getUpvoteState(
+    UserIdFirestore? userId,
+    ParentIdFirestore? parentId, {
+    Transaction? transaction,
+  }) {
+    return super.noSuchMethod(
+      Invocation.method(
+        #getUpvoteState,
+        [userId, parentId],
+        {#transaction: transaction},
```

Comment:
It is maybe a bit picky 😄 , but it may be better to keep the consistency of not-named parameter.
```suggestion
        [userId, parentId, transaction],
```

Reply:
Good suggestion !
However I think we might want to keep it like this to be closer to the real prototype of the function which is:
```dart
Future<UpvoteState> getUpvoteState(
    UserIdFirestore userId,
    ParentIdFirestore parentId, {
    Transaction? transaction,
  })
```
What do you think ?

Reply:
You are right 💯 that is a good idea.